src/Raven.Server/Web/RequestHandler.cs

```diff
@@ -408,7 +409,8 @@ protected bool TryGetServerAdmin()
                 case RavenServer.AuthenticationStatus.Expired:
                 case RavenServer.AuthenticationStatus.Allowed:
                 case RavenServer.AuthenticationStatus.NotYetValid:
-                    if (Server.Configuration.Security.AuthenticationEnabled == false)
+                    if (Server.Configuration.Security.AuthenticationEnabled == false 
+                        && SecurityUtils.IsUnsecuredAccessAllowedForAddress(Server.Configuration.Security.UnsecuredAccessAddressRange, HttpContext.Connection.RemoteIpAddress))
```

Comment:
OK, I thought apart from server binding validation, we're checking if client comes from proper zone.